src/lib/pdfconverter.cc

```diff
@@ -546,13 +546,26 @@ void PdfConverterPrivate::findLinks(QWebFrame * frame, QVector<QPair<QWebElement
 		uexternal  = PageObject::webPageToObject[frame->page()]->settings.useExternalLinks;
 	}
 	if (!ulocal && !uexternal) return;
-	foreach (const QWebElement & elm, frame->findAllElements("a")) {
+	foreach (QWebElement elm, frame->findAllElements("a")) {
 		QString n=elm.attribute("name");
 		if (n.isEmpty()) n=elm.attribute("ns0:name");
 		if (n.startsWith("__WKANCHOR_")) anchors[n] = elm;
 
 		QString h=elm.attribute("href");
-		if (h.isEmpty()) h=elm.attribute("ns0:href");
+		if (h.isEmpty()) {
+			h=elm.attribute("ns0:href");
+
+			//Internal links (anchor tags) are not linked correctly
+			//when the anchor's content is empty. This adds a space with
+			//a height of 0 when it is empty so that internal links
+			//are linked correctly.
+			if(elm.firstChild().isNull()){
```

Comment:
Okay, so then this approach is wrong/incomplete. The `ns0:href` is generated by the TOC support, so it is quite possible that it won't work there. A better approach is to check directly for the condition that you want (`<a name="not_empty"></a>`) rather than the absence of something (missing `href` on `<a>`) and work on that directly. Additionally, you might want to check for `id` tags on headers/divs and add some content there too (they can also be used as the target of an href).